src/elements/earn/pools/addLiquidity/empty/AddLiquidityEmptyCTA.tsx

```diff
@@ -30,46 +31,43 @@ export const AddLiquidityEmptyCTA = ({
 }: Props) => {
   const dispatch = useDispatch();
   const { account } = useWeb3React();
+  const history = useHistory();
 
   const handleAddLiquidity = async () => {
-    try {
-      const data = [
-        { amount: amountBnt, token: bnt },
-        { amount: amountTkn, token: tkn },
-      ];
-      const txHash = await addLiquidity(data, pool.converter_dlt_id);
-      dispatch(
-        addNotification({
-          type: NotificationType.pending,
-          title: 'Add Liquidity',
-          msg: `You added ${prettifyNumber(amountTkn)} ${
-            tkn.symbol
-          } and ${prettifyNumber(amountBnt)} ${bnt.symbol} to pool ${
-            pool.name
-          }`,
+    const data = [
```

Comment:
this entire approach of saving the current data to the closure and passing it to be called afterwards - is a risk that when it is called it might be already stale. This is a lot of high level logic and it makes sense to me that it will happen as a side effect of an action - using the most up to date info from redux.
we are doing too much things in components and need to move away from this practice. We are creating complex callbacks that carry a lot of data with them instead of simple callbacks that, when fired, read the latest data and use it.

Reply:
I agree